### PR TITLE
Improve build and deploy env definitions

### LINKED ISSUE #8815: Improve build and deploy env definitions
# General issue

Feedback on page: /cloud/project/magento-env-yaml.html

Recommend adding a note to these items, in particular ``deploy``:

```
o build—Controls actions in the build phase only. If you do not specify settings in this section, the build phase uses settings from the global section.
o deploy—Controls actions in the deploy phase only. If you do not specify settings in this section, the deploy phase uses settings from the global section.
```

## Possible solutions:

If you have enabled [Configuration Management](https://devdocs.magento.com/cloud/live/sens-data-over.html) after dumping the configurations, you should move the SCD_* variables from the deploy to the build stage.

E.g.,

Before Configuration Management:

```
  deploy:
    CRON_CONSUMERS_RUNNER:
      cron_run: true
      consumers: []
    SCD_STRATEGY: compact
    SCD_MATRIX:
      ...
    REDIS_USE_SLAVE_CONNECTION: 1
```

After enabling Configuration Management, move the SCD_* variables to the build stage:

```
  deploy:
    CRON_CONSUMERS_RUNNER:
      cron_run: true
      consumers: []
    REDIS_USE_SLAVE_CONNECTION: 1
  build:
    SCD_STRATEGY: compact
    SCD_MATRIX:
      ...
```

### PR DESCRIPTION
**Pull request by BarnyShergold**
## Purpose of this pull request

This pull request (PR) fixes #8815

## Affected DevDocs pages

https://devdocs.magento.com/cloud/project/magento-env-yaml.html
https://devdocs.magento.com/cloud/live/sens-data-over.html

whatsnew
Added [recommendations for improving build and deploy environment definitions](https://devdocs.magento.com/cloud/project/magento-env-yaml.html#configuration-management-optimization) related to [SCD performance](https://devdocs.magento.com/cloud/live/sens-data-over.html#scd-performance) if you have Configuration management enabled.

### CODE REVIEW

**Review by hguthrie — CHANGES_REQUESTED** [review]
Looks good in the SCD Performance section.
Not so keen on the addition to the File structure section.

If you duplicate the content and one set changes, then it stands to reason they both might require the same change. If there must be two instances of this information, then I recommend creating an `_include` file, that way if you change one, you change them all.

File: src/cloud/project/magento-env-yaml.md [hunk 1]
```diff
@@ -22,6 +22,38 @@ The `.magento.env.yaml` file includes the following sections:
    -  `email`—Configure an email to send to one or more email recipients.
    -  [logging handlers]({{ site.baseurl }}/cloud/env/log-handlers.html)—Configure hardware and software application messages sent to a remote logging server.
 
+{:.bs-callout-info}
+If you have enabled Configuration Management after dumping the configurations, you should move the SCD_* variables from the deploy to the build stage.
```
[the remark this answers is not in the document]
Comment (BarnyShergold) [reply src/cloud/project/magento-env-yaml.md]: I'm a  bit confused then as to your comment on the issue - you said "Good suggestions for improving the build and deploy definitions".

However, I've moved it to a new section and moved the content to an include.

[the remark this answers is not in the document]
Comment (hguthrie) [reply src/cloud/project/magento-env-yaml.md]: Yes, they are good suggestions...in the SCD Performance section of the Configuration management for store settings topic.
The problem is that it seemed out of the context or spirit of the **File structure** paragraph in the Build and deploy topic. That paragraph was defining the sections that make up the .magento.env.yaml file.

The new section looks great and is more focused. This way, if there is any further Config Management things to discuss in relation to the EnV config yaml, there is a clear place for it.

**Comment by dobooth** [comment]
@magento import pr to magento-devdocs/devdocs

**Comment by magento-engcom-team** [comment]
@dobooth the pull request successfully imported.

**Pull request merged**